starknet_py/contract.py

```diff
@@ -381,10 +416,74 @@ async def from_address(
         :raises BadRequest: when contract is not found
         :param address: Contract's address
         :param client: Client used
+        :param proxy_config: Proxy resolving config
         :return: an initialized Contract instance
         """
+        if proxy_config is None:
+            proxy_config = {}
+
+        max_steps = proxy_config.get("max_steps", 5)
+        proxy_checks = proxy_config.get("proxy_checks", False)
+
+        if proxy_checks is True:
+            proxy_checks = [ArgentProxyCheck(), OpenZeppelinProxyCheck()]
+        elif proxy_checks is False:
+            proxy_checks = []
+
+        contract = await Contract._create_contract_from_address(
+            address=address,
+            client=client,
+            proxy_checks=proxy_checks,
+            max_steps=max_steps,
+            step=1,
+        )
+
+        return Contract(address=address, abi=contract.data.abi, client=client)
+
+    @staticmethod
+    async def _create_contract_from_address(
+        # pylint: disable=too-many-arguments
```

Comment:
The problem with too many arguments comes from the fact that this method is recursive and `steps` and `processed_address` have to be passed every iteration. I'm not sure if it would make much sense to extract this method.

Reply:
Contract being a part of the API, having those methods, also static ones doesn't help with the clarity of code. If we extracted this to a factory - you'd notice that you can again use instance fields again which is more OOP-like.

Reply:
Alright, I can write a factory for creating contracts from address.